src/tests/test_domain.C

```diff
@@ -36,6 +41,129 @@ Teuchos::RCP<Teuchos::ParameterList> atmosphereParams;
     double xmin,xmax,ymin,ymax;
 
 Teuchos::RCP<Epetra_Comm> comm;
+
+double get_double(std::istream& input)
+{
+    std::string line;
+    std::getline(input, line);
+    return std::strtod(line.c_str(), nullptr);
+}
+
+::testing::AssertionResult
+validateDomain(int qz)
+{
+   ::testing::AssertionResult result = ::testing::AssertionSuccess();
+
+    xmin = 286 * PI_ / 180;
+    xmax = 350 * PI_ / 180;
+    ymin =  10 * PI_ / 180;
+    ymax =  74 * PI_ / 180;
+
+    n        = 16;
+    m        = 16;
+    l        = 8;
+    periodic = false;
+    double hdim = 4000.0;
+    std::ifstream input;
+    double value;
+    double ref_value;
+
+    domain = Teuchos::rcp(new TRIOS::Domain(n, m, l, _NUN_,
+                                            xmin, xmax, ymin, ymax,
+                                            periodic, hdim, qz, comm));
+
+    input = std::ifstream("x.qz" + std::to_string(qz) + ".values");
```

Comment:
Can't these just be put in a single HDF5 file?

Reply:
Probably, this was the simplest thing for debugging, but I can update it to use a HDF5 file.

Reply:
I assumed it'd be easier using HDF5 (also if you want to test the distributed domain at some point). I think it also makes sure the right precision is used. If it's not easier just leave it like this.